test/steps/v2_steps.py

```diff
@@ -711,12 +883,228 @@ def expect_path(context, path):
+            approval_program = bytearray(f.read())
+    if clear_program == "none":
+        clear_program = None
+    elif clear_program:
+        with open(dir_path + "/test-harness/features/resources/" + clear_program, "rb") as f:
+            clear_program = bytearray(f.read())
+    if int(local_ints) == 0 and int(local_bytes) == 0:
+        local_schema = None
+    else:
+        local_schema = transaction.StateSchema(num_uints=int(local_ints), num_byte_slices=int(local_bytes))
+    if int(global_ints) == 0 and int(global_bytes) == 0:
+        global_schema = None
+    else:
+        global_schema = transaction.StateSchema(num_uints=int(global_ints), num_byte_slices=int(global_bytes))
+    if app_args == "none":
+        app_args = None
+    elif app_args:
+        app_args = split_and_process_app_args(app_args)
+    if foreign_apps == "none":
+        foreign_apps = None
+    elif foreign_apps:
+        foreign_apps = [int(app) for app in foreign_apps.split(",")]
+    if app_accounts == "none":
+        app_accounts = None
+    elif app_accounts:
+        app_accounts = [account_pubkey for account_pubkey in app_accounts.split(",")]
+    application_id = 0
+    if hasattr(context, "current_application_id") and context.current_application_id:
+        application_id = context.current_application_id
+    sp = context.app_acl.suggested_params()
+    context.app_transaction = transaction.ApplicationCallTxn(sender=context.transient_pk, sp=sp,
+                                                             index=int(application_id),
+                                                             on_complete=operation, local_schema=local_schema,
+                                                             global_schema=global_schema,
+                                                             approval_program=approval_program,
+                                                             clear_program=clear_program,
+                                                             app_args=app_args, accounts=app_accounts,
+                                                             foreign_apps=foreign_apps,
+                                                             note=None, lease=None, rekey_to=None)
+
+
+@given('I sign and submit the transaction, saving the txid. If there is an error it is "{error_string}".')
+def sign_submit_save_txid_with_error(context, error_string):
+    signed_app_transaction = context.app_transaction.sign(context.transient_sk)
+    context.app_txid = context.app_acl.send_transaction(signed_app_transaction)
+
+
+@given('I sign and submit the transaction, saving the txid. If there is an error it is "".')
+def step_impl(context):
+    sign_submit_save_txid_with_error(context, "")
+
+
+@given('I wait for the transaction to be confirmed.')
+def step_impl(context):
+    sp = context.app_acl.suggested_params()
+    context.most_recent_pending_info = context.acl.pending_transaction_info(context.app_txid)
```

Comment:
Move this after `status_after_block`. There is no application ID because it is still pending at this point.